### PR TITLE
Extend unlock times for only airdrop users

### PR DESCRIPTION
**Pull request by and9477**

### CODE REVIEW

File: src/RewardLockerV3.sol [hunk 1]
```diff
@@ -0,0 +1,351 @@
...
+
+    function extendAirdropUnlockTime(uint256[] memory firstEntryIds, address[] memory users)
+        external
+        onlyUpdateUnlockTimeRole
+    {
+        require(firstEntryIds.length == users.length, "RewardLocker: mismatch lengths");
+        uint256 length = firstEntryIds.length;
+        for (uint256 i; i < length; i++) {
+            for (uint256 j; j < 26; j++) {
+                uint256 unlockTime = rewardEntries[firstEntryIds[i] + j][users[i]].unlockTime;
```
Comment (athos-dev1) [inline src/RewardLockerV3.sol:336]: can you check it's not 0

Comment (athos-dev1) [reply src/RewardLockerV3.sol:336]: and can we check that the user address stays the same? or it doesn't really matter?

File: src/RewardLockerV3.sol [hunk 2]
```diff
@@ -0,0 +1,351 @@
...
+
+        return (params.amount1Left.sub(currentAmountTo1), params.amount2Left.sub(currentAmountTo2));
+    }
+
+    function extendAirdropUnlockTime(uint256[] memory firstEntryIds, address[] memory users)
+        external
+        onlyUpdateUnlockTimeRole
+    {
+        require(firstEntryIds.length == users.length, "RewardLocker: mismatch lengths");
+        uint256 length = firstEntryIds.length;
```
Comment (athos-dev1) [inline src/RewardLockerV3.sol:333]: you can move this line up and then do `require(length == users.length, "RewardLocker: mismatch lengths");`

File: src/RewardLockerV3.sol (hunk 1 above)
Comment (athos-dev1) [inline src/RewardLockerV3.sol:336]: is this uint256 or uint40?

File: src/RewardLockerV3.sol [hunk 3]
```diff
@@ -0,0 +1,351 @@
...
+        for (uint256 i; i < length; i++) {
+            for (uint256 j; j < 26; j++) {
+                uint256 unlockTime = rewardEntries[firstEntryIds[i] + j][users[i]].unlockTime;
+                rewardEntries[firstEntryIds[i] + j][users[i]].unlockTime = uint40(unlockTime + 52 weeks);
+            }
+        }
+    }
+
+    function extendUnlockTime(uint256[] memory entryIds, address[] memory users) external onlyUpdateUnlockTimeRole {
```
Comment (athos-dev1) [inline src/RewardLockerV3.sol:343]: tbh it feels like you can just this function and not extendAirdropUnlockTime? it's not like you are doing anything special with extendAirdropUnlockTime besides looping 0-25

Comment (athos-dev1) [reply src/RewardLockerV3.sol:343]: or do you want to save some calldata by using a 0-25 loop instead of providing them one by one?

Comment (and9477) [reply src/RewardLockerV3.sol:343]: The purpose of using a 0 - 25 loop is that we only know the first entryID value per a user.
And to reduce the lengths of the parameters.

File: src/RewardLockerV3.sol (hunk 1 above)
Comment (and9477) [reply src/RewardLockerV3.sol:336]: unlockTime is a uint40 but I think it's okay to copy the value to uint256 for gas optimization.

**Pull request merged**